test/models/subnet_test.rb

```diff
@@ -102,13 +102,13 @@ class Subnet::Test < Subnet; end
 
   test "should strip whitespace before save" do
     s = subnets(:one)
-    s.network = " 10.0.0.22   "
+    s.network = " 10.0.22.0   "
     s.mask = " 255.255.255.0   "
     s.gateway = " 10.0.0.138   "
     s.dns_primary = " 10.0.0.50   "
     s.dns_secondary = " 10.0.0.60   "
     assert s.save
-    assert_equal "10.0.0.22", s.network
+    assert_equal "10.0.22.0", s.network
     assert_equal "255.255.255.0", s.mask
     assert_equal "10.0.0.138", s.gateway
```

Comment:
this gateway is not in 10.0.2.0/24, shouldn't it also be rejected? :)

Reply:
While I agree, I would not like to go crazy with this one. At least we should have ISC DHCP error about this one, not just REST API 404 that confuses Foreman.

Reply:
Yeah, but at least make the test use sensible data ;)

Reply:
Yeah it is easy to do, let me add this validation then.

Reply:
I don' think we need validation tbh. There's some edge cases like point to point connections. From my production `/etc/network/interfaces`:
```
auto eth0
iface eth0 inet static
        address 94.142.242.30/32
        gateway 94.142.244.11
        pointtopoint 94.142.244.11
```

This results in:
```
$ ip a show dev eth0
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000                                                         
    link/ether 52:54:00:09:52:90 brd ff:ff:ff:ff:ff:ff
    inet 94.142.242.30/32 brd 94.142.242.30 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 2a02:898:210::30/64 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::5054:ff:fe09:5290/64 scope link
       valid_lft forever preferred_lft forever
$ ip r
default via 94.142.244.11 dev eth0 onlink
```

Reply:
Brrrr. But yeah, that's valid (and a crude way to save IP addresses).

I really just meant to update the test to use a `10.0.22.0/24` gateway, that's all.

Reply:
:+1: to that

Reply:
Yeah missed that.